### PR TITLE
[REF] refactor fetch localizer contrasts

### PR DESCRIPTION
**Pull request by Remi-Gau**
- Closes none but relates to #3655 

Changes proposed in this pull request:

- extract input checks

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -227,10 +227,10 @@ convention = "numpy"
 
 [tool.ruff.lint.pylint]
 # https://docs.astral.sh/ruff/settings/#lint_pylint_max-args
-max-args = 26
+max-args = 27
```
Comment (Remi-Gau) [inline pyproject.toml:230]:
```suggestion
max-args = 26
```

File: nilearn/datasets/func.py [hunk 2]
```diff
@@ -902,6 +898,24 @@ def fetch_localizer_contrasts(
     return Bunch(ext_vars=csv_data, description=fdescr, **files)
 
 
+def _check_inputs_fetch_localizer_contrasts(contrasts):
```
Comment (bthirion) [inline nilearn/datasets/func.py:901]: I like to have one-line docstrings on private functions.

**Review by bthirion — COMMENTED** [review]
LGTM otherwise

File: nilearn/datasets/func.py [hunk 3]
```diff
@@ -902,6 +898,24 @@ def fetch_localizer_contrasts(
...
+            "Contrasts should be a list of strings, but "
+            f'a single string was given: "{contrasts}"'
+        )
+    unknown_contrasts = [
+        x
+        for x in contrasts
+        if (x not in ALLOWED_CONTRASTS and x not in CONTRAST_NAME_WRAPPER)
+    ]
+    if unknown_contrasts:
+        raise ValueError(
```
Comment (man-shu) [inline nilearn/datasets/func.py:913]: Maybe a couple of tests for the ValueErrors here? Or would that be out-of-scope for the PR?

File: nilearn/datasets/func.py [hunk 4]
```diff
@@ -784,33 +781,32 @@ def fetch_localizer_contrasts(
...
+                "/localizer",
+                "derivatives",
+                "spm_1st_level",
+                f"sub-{subject_id}",
+                (
+                    f"sub-{subject_id}_task-localizer"
+                    f"_acq-{contrast}_{data_type}.nii.gz"
+                ),
+            ]
+        )
```
Comment (man-shu) [inline nilearn/datasets/func.py:803]: I am wondering if this can also be refactored to a Path object?

**Review by man-shu — APPROVED** [review]
Otherwise LGTM!

File: nilearn/datasets/func.py (hunk 4 above)
Comment (Remi-Gau) [reply nilearn/datasets/func.py:803]: I think I tried this in the past and there was a good reason not to this, but I cannot remember.

In any case I would say this is beyond the scope of the PR.

Comment (Remi-Gau) [reply nilearn/datasets/func.py:803]: but feel free to try it in another PR. 😜

Comment (man-shu) [reply nilearn/datasets/func.py:803]: > I think I tried this in the past and there was a good reason not to this, but I cannot remember.

Looked at it again and it looks like we are trying to make a URL not a path so Path might give weird URL on windows machine

**Pull request merged**